Fix failing mypy linting

https://github.com/brettcannon/gidgethub/runs/714221145?check_suite_focus=true#step:6:155

```
gidgethub/httpx.py:24: error: unused 'type: ignore' comment
```

Either come up with some other way to deal with the http type issue (cast?), or stop using `--strict`.

--- a/gidgethub/httpx.py
+++ b/gidgethub/httpx.py
@@ -18,10 +18,7 @@ class GitHubAPI(gh_abc.GitHubAPI):
         response = await self._client.request(
             method,
             url,
-            # Reason: see discussion at https://github.com/brettcannon/gidgethub/pull/122#issuecomment-633738024.
-            # httpx 0.13.1 updated the HeaderTypes and it is acting invariant, causing our Mapping[str, str] to be
-            # technically incompatible with their Mapping[Union[str, bytes], Union[str, bytes]].
-            headers=dict(headers),  # type: ignore
+            headers=dict(headers),
             data=body,
         )
         return response.status_code, response.headers, response.content

--- a/gidgethub/httpx.py
+++ b/gidgethub/httpx.py
@@ -15,12 +15,7 @@ class GitHubAPI(gh_abc.GitHubAPI):
         self, method: str, url: str, headers: Mapping[str, str], body: bytes = b""
     ) -> Tuple[int, Mapping[str, str], bytes]:
         """Make an HTTP request."""
-        response = await self._client.request(
-            method,
-            url,
-            headers=dict(headers),
-            data=body,
-        )
+        response = await self._client.request(method, url, headers=dict(headers), data=body)
         return response.status_code, response.headers, response.content
 
     async def sleep(self, seconds: float) -> None:

--- a/gidgethub/httpx.py
+++ b/gidgethub/httpx.py
@@ -15,7 +15,9 @@ class GitHubAPI(gh_abc.GitHubAPI):
         self, method: str, url: str, headers: Mapping[str, str], body: bytes = b""
     ) -> Tuple[int, Mapping[str, str], bytes]:
         """Make an HTTP request."""
-        response = await self._client.request(method, url, headers=dict(headers), data=body)
+        response = await self._client.request(
+            method, url, headers=dict(headers), data=body
+        )
         return response.status_code, response.headers, response.content
 
     async def sleep(self, seconds: float) -> None:

